primitives/src/lib.rs

```diff
@@ -498,26 +498,74 @@ pub enum CurrencyId {
 	Token(u64),
 }
 
-impl CurrencyId {
-	pub const StellarNative: CurrencyId = Self::Stellar(Asset::StellarNative);
+pub trait DecimalsLookup {
+	type CurrencyId;
 
-	pub fn decimals(&self) -> u8 {
-		match self {
+	fn decimals(currency_id: CurrencyId) -> u32;
+
+	fn one(currency_id: CurrencyId) -> Balance {
+		10u128.pow(Self::decimals(currency_id))
+	}
+}
+
+// We use the PendulumDecimalsLookup as the default implementation in Spacewalk because it
+// is more interesting with the XCM(0) token having 10 decimals vs the default 12 decimals.
+pub type DefaultDecimalsLookup = PendulumDecimalsLookup;
+
+pub struct PendulumDecimalsLookup;
+impl DecimalsLookup for PendulumDecimalsLookup {
+	type CurrencyId = CurrencyId;
+
+	fn decimals(currency_id: CurrencyId) -> u32 {
+		(match currency_id {
 			CurrencyId::Stellar(asset) => asset.decimals(),
+			CurrencyId::XCM(index) => match index {
```

Comment:
I did not check whether this is correct.

Reply:
I adopted the decimals from the transactions we created for the asset metadata [here](https://github.com/pendulum-chain/transaction-builder/blob/9c8be6de9b9e0b92ad6150e8b8ee63095894800a/transactions/amplitude_transfer21.js#L74) and [here](https://github.com/pendulum-chain/transaction-builder/blob/9c8be6de9b9e0b92ad6150e8b8ee63095894800a/transactions/pendulum_transfer20.js#L17).